src/ngraph/runtime/gpu/gpu_external_function.cpp

```diff
@@ -254,6 +254,31 @@ runtime::gpu::GPU_ExternalFunction::~GPU_ExternalFunction()
 {
 }
 
+void runtime::gpu::GPU_ExternalFunction::assemble_and_reserve()
+{
+    GPUAllocator allocator = m_shared_context->m_primitive_emitter->get_memory_allocator();
+
+    for (shared_ptr<Function> current_function : m_pass_manager.get_state().get_functions())
+    {
+        // cache list of ordered ops
+        m_function_ordered_ops.insert({current_function, current_function->get_ordered_ops()});
+
+        // reserve memory buffers for intermediate results
+        size_t temp_pool_size = current_function->get_temporary_pool_size();
```

Comment:
this should go to `emit_temp_mem_pool_allocation`,

Reply:
The reservations for the pool need to be done prior to the op emissions. The function bodies are emitted in `collect_unique_functions` which occurs prior to `emit_temp_mem_pool_allocation`. By splitting up the reservation and the invocation in the way I have, the correct order is ensured.

I think I agree with @rkimballn1 now that I look at it. After separating out the reservation and the invocation, it looks like it would work naturally as a pass.

Reply:
Make sense.